src/OpenTelemetry/Internal/OpenTelemetrySdkEventSource.cs

```diff
@@ -105,7 +105,13 @@ public void ActivityStarted(Activity activity)
         {
             if (this.IsEnabled(EventLevel.Verbose, EventKeywords.All))
             {
-                this.ActivityStarted(activity.OperationName, activity.Id);
+                // Accessing activity.Id here will cause the Id to be initialized
+                // before the sampler runs. This will result in Id not reflecting the
```

Comment:
this comment is bit misleading, as this is only applicable to legacy Activity only.